modelci/app/experimental/endpoints/trainer.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Author: Li Yuanming
+Email: yli056@e.ntu.edu.sg
+Date: 1/15/2021
+"""
+from typing import List
+
+from fastapi import APIRouter
+
+from modelci.experimental.model.model_train import TrainingJob
+
+router = APIRouter()
+
+
+@router.post('/')
+def create_training_job(training_job: TrainingJob):
+    """
+    Create a training job data object, and save it into the database. Then, submit the created training job
+    (with job ID generated by database) to the training job coordinator.
+
+    Args:
+        training_job (TrainingJob): Training job to be submitted.
+
+    Returns:
+        Submitted training job data class object.
```

Comment:
As there is so many validator to the submitted data, maybe we should return the validation check result to indicate failure on creating a training job (or we should also write the validation rules in the frontend ? )

Reply:
This can be done via a catch in the API. Currently, we can assume all user inputs are correct